### PR TITLE
Fix logging crash

### PR DESCRIPTION
**Pull request by aiwantaozi**
Problem:
Get crash when only create project logging

Solution:
Check before directly use the clusterLoggingList

Issue:
https://github.com/rancher/rancher/issues/17606

### CODE REVIEW

File: pkg/controllers/user/logging/configsyncer/configsyncer.go [hunk 1]
```diff
@@ -137,7 +137,12 @@ func (s *ConfigSyncer) syncClusterConfig(clusterLoggings []*mgmtv3.ClusterLoggin
 	secretName := loggingconfig.RancherLoggingConfigSecretName()
 	namespace := loggingconfig.LoggingNamespace
 
-	buf, err := s.configGenerator.GenerateClusterLoggingConfig(clusterLoggings[0], systemProjectID)
+	var clusterLogging *mgmtv3.ClusterLogging
+	if len(clusterLoggings) != 0 {
```
Comment (orangedeng) [inline pkg/controllers/user/logging/configsyncer/configsyncer.go]: If there is nothing for input, it should be return earlier. I would like to fix this in line 111. `If len(clusterLoggings) = 0`, `syncClusterConfig` won't be called.

Comment (aiwantaozi) [reply pkg/controllers/user/logging/configsyncer/configsyncer.go]: when the (clusterLoggings) == 0, it need to update the secret to empty config

Comment (orangedeng) [reply pkg/controllers/user/logging/configsyncer/configsyncer.go]: I got it now. We need to have a empty `[]byte` to update the configuration secret.

**Comment by orangedeng** [comment]
LGTM

**Pull request merged**